drivers/pinmux/pinmux_stm32.c

```diff
@@ -46,15 +46,26 @@ const struct device * const gpio_ports[STM32_PORTS_MAX] = {
 static int stm32_pin_configure(uint32_t pin, uint32_t func, uint32_t altf)
 {
 	const struct device *port_device = gpio_ports[STM32_PORT(pin)];
+	int ret = 0;
 
-	/* not much here, on STM32F10x the alternate function is
-	 * controller by setting up GPIO pins in specific mode.
-	 */
-	if (port_device == NULL) {
+	if ((port_device == NULL) || (!device_is_ready(port_device))) {
 		return -ENODEV;
 	}
 
-	return gpio_stm32_configure(port_device, STM32_PIN(pin), func, altf);
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+	ret = pm_device_get(port_device);
+#endif
+	if (ret != 0) {
+		return ret;
+	}
+
+	gpio_stm32_configure(port_device, STM32_PIN(pin), func, altf);
```

Comment:
We lose the "return status" of `gpio_stm32_configure()` ?

Reply:
There is no error checking done in this function.
I wonder if it shouldn't be set to void.

Reply:
Function moved to void